src/main/java/seedu/duke/parsers/EditParser.java

```diff
@@ -26,23 +34,68 @@
     // \s+-d\s+\"[^\"]+\"|\s+-t\s+\"[^\"]+\")(\s+-n\s+\"((?<taskName>[^\"]+)\")?|\s+-d\s+\"
     // ((?<taskDescription>[^\"]+)\")?|(\s+-t\s+\"(?<estimatedWorkingTime>[^\"]+)\")?))|(mod\s+
     // (?<moduleCode>\w+?(?=(\s+-d\s+)))(\s+(-d\s+\"(?<moduleDescription>.+)\"))))(?<invalid>.*)
-    private static final String EDIT_FORMAT = "((task\\s+(?<taskNumber>\\d+)(\\s+-m\\s+(?<taskModule>\\w+))?"
-            + "(?=\\s+-n\\s+\\\"[^\\\"]+\\\"|\\s+-d\\s+\\\"[^\\\"]+\\\"|\\s+-t\\s+\\\"[^\\\"]+\\\")(\\s+-n\\s+\\\""
-            + "((?<taskName>[^\\\"]+)\\\")?|\\s+-d\\s+\\\"((?<taskDescription>[^\\\"]+)\\\")?|(\\s+-t\\s+\\\""
-            + "(?<estimatedWorkingTime>[^\\\"]+)\\\")?))|(mod\\s+(?<moduleCode>\\w+?(?=(\\s+-d\\s+)))(\\s+(-d\\s+\\\""
-            + "(?<moduleDescription>.+)\\\"))))(?<invalid>.*)";
+
+    /* Explanation for regex:
+     * ((task\s+(?<taskNumber>\d+|(?<invalidNumber>.*))      -- matches [task taskNumber].
+     *
+     * (\s+(-m|(?<invalidModFlag>.*))\s+(?<taskModule>\w+))? -- matches [-m taskModule] if present. Optional
+     *                                                          Note that taskModule does not require "", but must be a
+     *                                                          single word composed of [a-zA-Z0-9_].
+     *
+     * (?=\s+-n\s+\"[^\"]+\"|\s+-d\s+\"[^\"]+\"              -- matches flags -n, -d, -t.
+     * |\s+-t\s+\"[^\"]+\")
+     *
+     * (\s+(-n|(?<invalidTaskNameFlag>.*))                   -- matches [-n "taskName"] or
```

Comment:
correct me if im wrong, but isnt the issue here because either `-d` or `-t` entered will be flagged as invalid because the regex matches the group in order and `(?<invalidTaskNameFlag>.*)` captures anything that is not `-n`. Im not sure how do you want to make the error message more descriptive since the parameter here can be either `-d`, `-n` or `-t`, so unless the user has specified `-d` and then entered something invalid, only then can we conclude that the TASK_DESCRIPTION is wrong, so if that is the case, then if theres an error, then we can check which of the 3 optional parameters is used and give the error message as such, or if all 3 optional parameters are not captured, then it means the flag is incorrect

Reply:
> So if that is the case, then if theres an error, then we can check which of the 3 optional parameters is used and give the error message as such, or if all 3 optional parameters are not captured, then it means the flag is incorrect

Not in the regex but as logic i guess